tools/klee/main.cpp

```diff
@@ -1227,6 +1244,113 @@ void KleeHandler::dumpCallPath(const ExecutionState &state,
+  std::vector<std::string> time_fns = {"start_time", "restart_time", "current_time", "get_start_time_internal", "get_start_time", "clock_gettime", "gettimeofday"};
+  std::vector<std::string> verif_fns = {"loop_iteration_assumptions", "loop_iteration_assertions", "loop_invariant_consume", "loop_invariant_produce", "loop_iteration_begin", "loop_iteration_end", "loop_enumeration_begin", "loop_enumeration_end", "allocate_unique_name", "count_reuse", "init_test_data", "report_internal_error", "rand_byte", "bridge_loop_invariant_consume", "bridge_loop_invariant_produce", "bridge_loop_iteration_begin", "bridge_loop_iteration_end", "bridge_loop_iteration_assumptions", "nf_loop_iteration_begin", "nf_add_loop_iteration_assumptions", "nf_loop_iteration_end", "concretize_devices", "flow_consistency", "rte_eth_dev_count", "flood", "exit", "__uClibc_fini", "_stdio_term"};
+  std::regex symbol_re("klee*");
+  std::regex symbol2_re("_exit@plt*");
+
+  //Now we start iterating over the input trace and print only stuff we demarcate
+  int currently_demarcated = 0;
+  std::string currently_demarcated_fn = "";
+  std::string last_written_fn = "";
+  int call_gap = 1;
+  for(auto it: state.stackInstrMap){
+    std::vector<std::string> call_stack = it.first;
+    std::string opcode = it.second->getOpcodeName();
+    int s = it.first.size();
+    if(s==0){
+      continue; // Cant do anything with an empty call stack
+    }
+    std::string current_fn_name = it.first[s-1];
+    int stateful = 0;
+    int dpdk = 0;
+    int time = 0;
+    int verif = 0;
+
+    if(currently_demarcated){
+      if(opcode != "ret"){
+        continue;
+      }
+    }
+
+    if(currently_demarcated && opcode == "ret" && current_fn_name == currently_demarcated_fn){
+      currently_demarcated = 0;
+      call_gap = 0;
+      currently_demarcated_fn = "";
+    }
+    else if(currently_demarcated == 0){
+      std::string fn_name;
+      for(auto it1 : call_stack){
+        fn_name = it1;
+        if(std::find(stateful_fns.begin(), stateful_fns.end(), it1) != stateful_fns.end()){
+          stateful = 1;
+          break;
+        }
+        else if(std::find(dpdk_fns.begin(), dpdk_fns.end(), it1) != dpdk_fns.end()){
+          dpdk = 1;
+          break;
+        }
+        else if(std::find(time_fns.begin(), time_fns.end(), it1) != time_fns.end()){
+          time = 1;
+          break;
+        }
+        else if(std::find(verif_fns.begin(), verif_fns.end(), it1) != verif_fns.end() || std::regex_match(it1, symbol_re) || std::regex_match(it1, symbol2_re)){
+          verif = 1;
+          break;
+        }
+      }
+
+      if(stateful or dpdk or time or verif){
+        currently_demarcated = 1;
+        currently_demarcated_fn = current_fn_name;
+      }
+
+      if(stateful || dpdk || verif || time){
+        if(call_gap){
+          last_written_fn = currently_demarcated_fn;
+          if(stateful){
+            *file << "Call to libVig model - " << fn_name << "\n";
```

Comment:
I'd also change this. Make the third word a table entry and pull it from there so we have just one print statement